R/plot_variance.R

```diff
@@ -0,0 +1,66 @@
+#' Plot as a heatmap the scaled between imputation variance for every cell
+#'
+#' This function plots the cell-level between imputation variance. The function
+#' scales the variances column-wise and plots the data image as a heatmap.
+#' Darker red cells indicate more variance, lighter cells indicate less variance.
+#' white cells represent observed cells or unobserved cells with zero between
+#' imputation variance.
+#'
+#' @param object A `mice` generated multiply imputed data set of class `mids`.
+#'
+#' @return An object of class `ggplot`.
+#' @importFrom dplyr mutate
+#' @importFrom dplyr across
+#' @importFrom dplyr everything
+#' @importFrom ggplot2 aes
+#' @importFrom mice mdc
+#' @importFrom dplyr select
+#' @importFrom dplyr group_by
+#' @importFrom dplyr summarise
+#' @importFrom dplyr ungroup
+#' @importFrom tidyr pivot_longer
+#' @importFrom stats var
+#' @examples
+#' library(mice)
+#' library(dplyr)
+#' library(tidyr)
+#' library(ggplot2)
+#' imp <- mice(nhanes)
+#' plot_variance(imp)
+#' @export
+plot_variance <- function(object){
+  if (!mice::is.mids(object)) {
+    stop("Input is not a Multiply Imputed Data Set of class mids. \n
+         Perhaps function mice::as.mids() can be of use?")
+  }
+  gg <- mice::complete(object, "long")  %>%
+    dplyr::mutate(across(where(is.factor), as.numeric)) %>%
+    dplyr::select(-.imp) %>%
+    dplyr::group_by(.id) %>%
+    dplyr::summarise(across(everything(),  stats::var)) %>%
+    dplyr::ungroup() %>%
+    mutate(across(.cols = -.id, ~ scale_above_zero(.))) %>%
```

Comment:
Please use double colon notation for `mutate()` and `across()`

Reply:
If the CMD check throws a NOTE about `.id` being undefined, please add it in the utils.R file under:
```
# suppress undefined global functions or variables note
utils::globalVariables(c(".imp", ".where"))
```